### PR TITLE
PHP 8.0 Support

### PR DESCRIPTION
**Pull request by diwms**
Hi there 👋🏻
I'm trying to contribute and close issue #2

I've no idea what I need to write in the issue template since this is an attempt to close the existing issue 😄

Please, let me know what do you think guys 👍🏻

Happy Hacktoberfest! 🥇

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,30 +6,25 @@ cache:
...
+    - php: 7.4
       env:
         - DEPS=latest
-    - php: 7.3
-      env:
-        - DEPS=lowest
-    - php: 7.3
+    - php: nightly
       env:
         - DEPS=latest
```
Comment (weierophinney) [inline .travis.yml:35]: There should be entries for both lowest and latest deps.

File: composer.json [hunk 2]
```diff
@@ -26,15 +26,16 @@
         }
     },
     "require": {
-        "php": "^7.1",
+        "php": "^7.3 || ^8.0",
```
Comment (weierophinney) [inline composer.json]: Please follow the brief as presented in #2 - the additional constraint should be `~8.0.0`.

File: composer.json [hunk 3]
```diff
@@ -26,15 +26,16 @@
...
         "laminas/laminas-zendframework-bridge": "^1.0",
         "mezzio/mezzio-authentication": "^1.0",
         "psr/container": "^1.0",
         "psr/http-message": "^1.0.1"
     },
     "require-dev": {
         "laminas/laminas-coding-standard": "~1.0.0",
-        "phpunit/phpunit": "^7.0.1"
+        "phpspec/prophecy-phpunit": "^2.0",
+        "phpunit/phpunit": "^9.4"
```
Comment (weierophinney) [inline composer.json]: Please see the brief in #2; this should be `^9.3`.

**Review by weierophinney — CHANGES_REQUESTED** [review]

**Comment by geerteltink** [comment]
@diwms Can you please start with a good short description for your commit. Instead of this:

> trying to close #2 …
> PHP8 support!

... do it the other way around:

> Add PHP 8 support
>
> Fixes #2

File: .travis.yml [hunk 4]
```diff
@@ -6,30 +6,25 @@ cache:
...
-    - php: 7.1
+    - php: 7.3
       env:
         - DEPS=lowest
-    - php: 7.1
+    - php: 7.3
       env:
         - DEPS=latest
-        - CS_CHECK=true
-        - TEST_COVERAGE=true
```
Comment (geerteltink) [inline .travis.yml:22]: - CS_CHECK=true
        - TEST_COVERAGE=true

Should both be added for another version.

Comment (diwms) [reply .travis.yml:22]: @geerteltink it's only for php8 or other versions too?
when wee need cs-check and coverage?

Comment (Ocramius) [reply .travis.yml:22]: @geerteltink I think it's safe to keep this on a single CI entry. PHP 8 is too unstable to run these checks there, for now.

File: .travis.yml [hunk 5]
```diff
@@ -6,30 +6,30 @@ cache:
 
 env:
   global:
-    - COMPOSER_ARGS="--no-interaction"
+    - COMPOSER_ARGS="--no-interaction --ignore-platform-reqs"
```
Comment (Ocramius) [inline .travis.yml]: `--ignore-platform-reqs` should really only happen for `php: nightly`: all other versions should **not** have this flag, or else we will end up with incompatible dependencies.

**Review by Ocramius — CHANGES_REQUESTED** [review]
LGTM :+1:

File: .travis.yml (hunk 4 above)
Comment (geerteltink) [reply .travis.yml:22]: I know, but in a previous commit it was entirely removed. With a force push you can't see the old commit history.

**Review by geerteltink — COMMENTED** [review]
There are multiple phpunit warnings about `Method ReflectionParameter::getClass() is deprecated`.

**Review by Ocramius — APPROVED** [review]
LGTM 👍

Thanks @diwms! Let's continue #3 now

**Pull request merged**